core/listener.go

```diff
@@ -101,11 +96,16 @@ func (cl *Listener) Start(context.Context) error {
 	cl.cancel = cancel
 	cl.closed = make(chan struct{})
 
-	sub, err := cl.fetcher.SubscribeNewBlockEvent(ctx)
+	err := cl.fetcher.Start(ctx)
+	if err != nil {
+		return err
+	}
+	subs, err := cl.fetcher.runSubscriber()
```

Comment:
nit suggestion: listener can pass subscriber context to fetcher. It will allow listener to manage lifecycle of subscription. fetcher on other hand would not have lifecycle at all.

Reply:
I like the idea of not managing the lifecycle of the fetcher.